### ISSUE TITLE
Multi GPUs Multi Nodes acceleration

### ISSUE THREAD

**Issue by WalkerRusher** [thuml/iVideoGPT#7]
Thx for your great work.

I try to use 4nodes x 8 GPUs to accelerate the training while using the following scripts:
accelerate launch train_tokenizer.py \
    --exp_name bair_tokenizer_ft --output_dir log_vqgan --seed 0 --mixed_precision bf16 \
    --model_type ctx_vqgan \
    --train_batch_size 16 --gradient_accumulation_steps 1 --disc_start 1000005 \
    --oxe_data_mixes_type bair --resolution 64 --dataloader_num_workers 16 \
    --rand_select --video_stepsize 1 --segment_horizon 16 --segment_length 8 --context_length 1 \
    --pretrained_model_name_or_path pretrained_models/ivideogpt-oxe-64-act-free/tokenizer

**yet the training time never be decreased compared to only using a single GPU.**

I have no ideas why this happened.

**Comment by WalkerRusher** [comment]
Did I miss anything? Any suggestions would be appreciated!

**Comment by Manchery** [comment]
In the script, we specify *per device* batch size as 16; thus, using more GPUs only makes the global batch size larger, but computation on each GPU is unchanged. If you want to accelerate training by more GPUs, you should make per device batch size smaller.

Hope this response resolves your question!

**Comment by WalkerRusher** [comment]
Thanks for your instant reply. 

Do you mean that IF I use more GPUs, for example, 8 GPUs, then I need to decrease the batch size to 2 in the scripts? After do that, do I still need training the tokenizer 1000000 steps the exactly reproduce the official results? 

Looking forward to your suggestions, deep thanks

**Comment by WalkerRusher** [comment]
And what if I want use 32 GPUs to accelerate the training, how should I modify the batch size and the training steps?

Thanks for your any suggestions.

**Comment by Manchery** [comment]
If you use N GPUs, you should reduce the batch size by N/4 (we use 4 GPUs), which maintains the same global batch size (16 per device x 4 devices for our experiments).

To be specific, If you want to use 8 (or 32) GPUs, you can set the batch size to 8 (or 2)

If you want to use GPUs more than 64, you can only set the batch size to 1, resulting in a global batch size larger than our specified 64. We do not implement more sophisticated parallel methods, such as model parallel, so one device should at least use 1 batch size.

**Comment by Manchery** [comment]
> Thanks for your instant reply.
> 
> Do you mean that IF I use more GPUs, for example, 8 GPUs, then I need to decrease the batch size to 2 in the scripts? After do that, do I still need training the tokenizer 1000000 steps the exactly reproduce the official results?
> 
> Looking forward to your suggestions, deep thanks

You only need to fine-tune the tokenizer approximately 10^5 steps to reproduce reported results on BAIR datasets. You can refer to Table 3 of our paper for detailed hyperparameters.

**Comment by WalkerRusher** [comment]
Thanks for your suggestions, it works well!
